File: foosball-stats-app/src/app/shared/models/event.ts
import { EventTeam } from "./eventTeam";
import { EventType } from "./eventType";

export interface Event {
  id: number;
  name: string;
  description: string;
  type: EventType;
  date: Date;
  teamsCount: number;
  teams: EventTeam[];
  relevantForRanking: boolean;
}

File: backend/Database/Entities/Event.cs
using System;
using System.Collections.Generic;
using Database.Enums;

namespace Database.Entities
{
    public class Event
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Url { get; set; }
        public string Description { get; set; }
        public EventType Type { get; set; }
        public DateTimeOffset Date { get; set; }
        public int TeamsCount { get; set; }
        public List<EventTeam> Teams { get; set; }
    }
}